dirsrvtests/tests/suites/replication/urp.py

```diff
@@ -0,0 +1,379 @@
+        3. Should Success.
+        4. Should Success.
+    """
+    log.info(f'actions order is: {actionorder}')
+    urp_tester.reset()
+    # Run the actions on specified server with specified order
+    for inst_idx,action in actionorder:
+        urp_tester.insts[inst_idx-1].run_action(action)
+    # Wait until replication is in sync
+    urp_tester.resync_all_agmts()
+    # Check that all replica have the sames entries
+    urp_tester.get_entries()
+    assert urp_tester.insts[0].entry_info == urp_tester.insts[1].entry_info
+    assert urp_tester.insts[0].entry_info == urp_tester.insts[2].entry_info
+
+
+def gen_ldif_file(first, second, result):
+    with open(result, 'w') as output:
+        for file in [ first, second ]:
+            with open(file, 'r') as input:
+                for line in input:
+                    output.write(line)
+
+
+
+@pytest.mark.skipif(SKIP_THE_TESTS, reason="This test is meant to execute in specific test environment")
+def test_urp_with_crossed_entries(topology_m2):
+    """Test urp behaviour if entry conflict entries are crossed
+
+    :id: 6b4adfc2-fd9b-11ee-a6f0-482ae39447e5
+    :setup: Two suppliers
+    :steps:
+        1. Generate ldif files with crossed conflict entries
+        2. Import the ldif files
+        3. Remove a conflict entry
+        4. Wait until le replication is in sync
+        5. Check that entries have the same type (tombstone/conflict/regular) on both replica.
+        6. Import the ldif files
+        7. Remove a regular entry
+        8. Wait until le replication is in sync
+        9. Check that entries have the same type (tombstone/conflict/regular) on both replica.
+    :expectedresults:
+        1. Should Success.
+        2. Should Success.
+        3. Should Success.
+        4. Should Success.
+    """
+    # Note in theory crossed entries should not happen
+    # But this test shows that we are in trouble if that is the case
+    s1 = topology_m2.ms["supplier1"]
+    s2 = topology_m2.ms["supplier2"]
+    datadir = os.path.join(os.path.dirname(__file__), '../../data/urp')
+    # 1. Generate ldif files with crossed conflict entries
+    # Export the ldif to be sure that the replication credentials are ok
+    export_ldif = f'{s1.get_ldif_dir()}/db.ldif'
+    import_ldif1 = f'{s1.get_ldif_dir()}/db1.ldif'
+    import_ldif2 = f'{s1.get_ldif_dir()}/db2.ldif'
+    Tasks(s1).exportLDIF(DEFAULT_SUFFIX, output_file=export_ldif, args={EXPORT_REPL_INFO: True, TASK_WAIT: True})
+    gen_ldif_file(export_ldif, f'{datadir}/db1.ldif', import_ldif1)
+    gen_ldif_file(export_ldif, f'{datadir}/db2.ldif', import_ldif2)
+    # 2. import the ldif files
+    Tasks(s1).importLDIF(DEFAULT_SUFFIX, input_file=import_ldif1, args={TASK_WAIT: True})
+    Tasks(s2).importLDIF(DEFAULT_SUFFIX, input_file=import_ldif2, args={TASK_WAIT: True})
+    # 3. Remove a conflict entry
+    ConflictEntry(s1, 'cn=u22449+nsUniqueId=c6654281-f11b11ee-ad93a02d-7ba2db25,dc=example,dc=com').delete()
```

Comment:
I have a failure here, through:

`E         ldap.OPERATIONS_ERROR: {'msgtype': 107, 'msgid': 88, 'result': 1, 'desc': 'Operations error', 'ctrls': [], 'ldap_request': "delete_ext_s(('cn=u22449+nsUniqueId=c6654281-f11b11ee-ad93a02d-7ba2db25,dc=example,dc=com',),{'serverctrls': None, 'clientctrls': None, 'escapehatch': 'i am sure'}) on instance supplier1"}`

When I run it standalone:

`[root@vm-10-0-186-102 ds]# URP_VERY_LONG_TEST=yes py.test-3 -s -v dirsrvtests/tests/suites/replication/urp.py::test_urp_with_crossed_entries &> rereres2`

Reply:
Although I know that this test is failing, your failure is unexpected:
I have a different behavior on my machine:

```
URP_VERY_LONG_TEST=yes py.test-3 -s -v $PWD/urp.py::test_urp_with_crossed_entries
...
>       assert u1 == u2
E       AssertionError: assert ['regular:cd8...02d-7ba2db25'] == ['conflict:cd...02d-7ba2db25']
...
```

Probably worth to try to understand why we get different results

Reply:
Recheck, and I got the same error as you now:

```
        u1 = search_entries(s1, 'cn=u22449')
        u2 = search_entries(s2, 'cn=u22449')
>       assert u1 == u2
E       AssertionError: assert ['regular:cd8...02d-7ba2db25'] == ['conflict:cd...02d-7ba2db25']
```

Reply:
Ok, this time it is the expected result.   URP does not properly handle this case 
BTW, I should add a @pytest.mark.xfail for this one so it will be explicit.